packages/zengine-sdk/src/filter-matcher.ts

```diff
@@ -0,0 +1,263 @@
+  return input
+    .replace('$', '')
+    .replace(',', '')
+}
+
+interface RuleMatcherInterface {
+  [key: string]: Function
+}
+
+export const matchers: RuleMatcherInterface = {
+  ruleEquals: function (recordValue: any, ruleValue: any): boolean {
+    if (Array.isArray(recordValue)) {
+      return recordValue.length === 0
+    }
+    return ruleValue.toString().toLowerCase() === recordValue.toString().toLowerCase()
+  },
+  ruleDoesNotEqual: function (recordValue: any, ruleValue: any): boolean {
+    if (Array.isArray(recordValue)) {
+      return recordValue.length > 0
+    }
+    return ruleValue.toString().toLowerCase() !== recordValue.toString().toLowerCase()
+  },
+  ruleMinimum: function (recordValue: any, ruleValue: any) {
+    if (Array.isArray(recordValue)) {
+      return false
+    }
+    if (recordValue === '') {
+      return false
+    }
+    // Do numeric comparison if filter rule value is numeric. For consistency with API,
+    // record values are somewhat forgiving, while filter rules must be a valid number.
+    if (ruleValue && !isNaN(Number(ruleValue))) {
+      recordValue = parseNumber(recordValue)
+
+      try {
+        ruleValue = parseFloat(ruleValue)
+        recordValue = parseFloat(recordValue)
+      } catch (err) {
+        return false
+      }
+
+      return recordValue >= ruleValue
+    }
+    // Loose comparison, don't use 'localeCompare' because we only want to compare
+    // numbers or well-formatted dates, which work with normal string comparison.
+    return recordValue >= ruleValue
+  },
+  ruleMaximum: function (recordValue: any, ruleValue: any) {
+    if (Array.isArray(recordValue)) {
+      return false
+    }
+    if (recordValue === '') {
+      return false
+    }
+    // Do numeric comparison if filter rule value is numeric. For consistency with API,
+    // record values are somewhat forgiving, while filter rules must be a valid number.
+    if (ruleValue && !isNaN(Number(ruleValue))) {
+      recordValue = parseNumber(recordValue)
+
+      try {
+        ruleValue = parseFloat(ruleValue)
+        recordValue = parseFloat(recordValue)
+      } catch (err) {
+        return false
+      }
+
+      return recordValue <= ruleValue
+
+    }
+    // Loose comparison, don't use 'localeCompare' because we only want to compare
+    // numbers or well-formatted dates, which work with normal string comparison.
+    return recordValue <= ruleValue
+  },
+  ruleContains: function (recordValue: any, ruleValue: any) {
+    if (Array.isArray(recordValue)) {
+
+      // Normalize Value as Array
+      if (!Array.isArray(ruleValue)) {
+        ruleValue = [ruleValue]
+      }
+
+      for (let index = 0; index < ruleValue.length; index++) {
+        if (recordValue.indexOf(ruleValue[index]) === -1) {
+          return false
+        }
+      }
+      return true
+    } else {
+      return recordValue.toString().indexOf(ruleValue.toString()) !== -1
```

Comment:
what if `recordValue == null`?

Reply:
i should have just left it as `String(recordValue)` instead of getting cute with `.toString()`... changes incoming

Reply:
as long as `String(null) // "null"` is also an acceptable outcome for this logic.

Reply:
also `String(undefined) // "undefined"`

Reply:
yeah both cases give an empty string and that's the original behavior from anglerfish